File: ThePeacenet.Backend/Manual/ManualPageBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ThePeacenet.Backend.Manual
{
    public class ManualPageBuilder
    {
        private ManualPage _page = new ManualPage();

		public string Id { get => _page.Id; set => _page.Id = value; }
        public string Name { get => _page.Name; set => _page.Name = value; }
        public string Description { get => _page.Description; set => _page.Description = value; }
        public string ItemType { get => _page.ItemType; set => _page.ItemType = value; }

        public ManualPage Page => _page;

		public void SetMetadata(string title, string content)
        {
			if(_page.Metadata.Any(x=>x.Title == title))
            {
                _page.Metadata.First(x => x.Title == title).Content = content;
            }
			else
            {
                _page.Metadata.Add(new ManualMetadata
                {
                    Title = title,
                    Content = content
                });
            }
        }
    }
}

File: Source/ProjectOglowia/Private/ManualPageBuilder.cpp
/********************************************************************************
 * The Peacenet - bit::phoenix("software");
 * 
 * MIT License
 *
 * Copyright (c) 2018-2019 Michael VanOverbeek, Declan Hoare, Ian Clary, 
 * Trey Smith, Richard Moch, Victor Tran and Warren Harris
 * 
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 * 
 * Contributors:
 *  - Michael VanOverbeek <alkaline@bitphoenixsoftware.com>
 *
 ********************************************************************************/

#include "ManualPageBuilder.h"

void UManualPageBuilder::SetItemType(const FText& InItemType) {
    this->ManualPage.ItemType = InItemType;
}

void UManualPageBuilder::SetID(FName InManualID) {
    this->ManualPage.ID = InManualID;
}

void UManualPageBuilder::SetFullName(const FText& InFullName) {
    this->ManualPage.FullName = InFullName;
}

void UManualPageBuilder::SetSummary(const FText& InSummary) {
    this->ManualPage.Summary = InSummary;
}

void UManualPageBuilder::SetMetadata(const FText& InMetaName, const FText& InMetaContent) {
    for(auto& Metadata : this->ManualPage.ManualMetadata) {
        if(Metadata.Title.EqualTo(InMetaName)) {
            Metadata.Content = InMetaContent;
            return;
        }
    }

    FManualMetadata NewMeta;
    NewMeta.Title = InMetaName;
    NewMeta.Content = InMetaContent;
    this->ManualPage.ManualMetadata.Add(NewMeta);
}

FManualPage UManualPageBuilder::GetManualPage() {
    return this->ManualPage;
}